core/commands/patch.go

```diff
@@ -0,0 +1,132 @@
+package commands
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	mh "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multihash"
+	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
+
+	cmds "github.com/ipfs/go-ipfs/commands"
+	u "github.com/ipfs/go-ipfs/util"
+)
+
+var PatchCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Create a new merkledag object based on an existing one",
+		ShortDescription: `
+'ipfs patch <root> [add-link|rm-link] <args>' is a plumbing command used to
+build custom DAG objects.
+
+Examples:
+
+	EMPTY_DIR=QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn 
```

Comment:
On Thu, May 28, 2015 at 10:38:40PM -0700, Jeromy Johnson wrote:
> its an example.

‘ipfs object new unixfs-dir’ reads more clearly to me, and I'd
certainly rather set a good example of using that function in help
text that might get copy-pasted into someone's script.  For example,
the ‘object new’ approach future-proofs us against changes to the
default unixfs-dir object (e.g. if we start tracking type information
in the object, ipfs/ipfs#36).